exchanges/binance/binance.go

```diff
@@ -72,6 +72,10 @@ type Binance struct {
 	validIntervals []TimeInterval
 }
 
+func (b *Binance) GetHistoricCandles(pair currency.Pair, rangesize int, granularity int) ([]exchange.Candle, error) {
+	return nil, common.ErrFunctionNotSupported
```

Comment:
Thanks for the explanation. Indeed it makes sense.
Question on the wrapper funcs, please. I understand GCT aims at providing some kind of unified API across all Exchanges, as far as it is possible.
So the layers are organised as follow:

- rpc.proto defines the GCT Unified API layer
- Then every exchange implements the IBot interface provides access to each Exchange
- the exchange wrapper layer is meant to transcode the data model between the unified API and the exchange peculiar API

Is that correct? I can see there are more details than what I listed, I am keen to understand the purpose of the wrapper layer at this moment. Cheers

Reply:
The GCT unified API is derived from the IBotInterface (https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/interfaces.go#L18). All exchanges have to implement these funcs which gives the engine (and bot) the ability to interact with each exchange through its wrapper funcs. Otherwise, for individual package usage (outside of the engine usage), the application can use any func inside of the exchanges package. 

The gRPC server runs as a subsystem via engine, which allows client -> server interaction to cause the engine to perform any actions you'd like on those exchanges (submitting/cancelling orders, getting info etc. all done through the exchange wrappers)